up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/Notifier.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.uprotocol.core.udiscovery;
+
+import static org.eclipse.uprotocol.common.util.log.Formatter.join;
+import static org.eclipse.uprotocol.common.util.log.Formatter.tag;
+import static org.eclipse.uprotocol.core.udiscovery.internal.Utils.toLongUri;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.SERVICE;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import org.eclipse.uprotocol.UPClient;
+import org.eclipse.uprotocol.common.util.log.Key;
+import org.eclipse.uprotocol.core.udiscovery.v3.Notification;
+import org.eclipse.uprotocol.core.udiscovery.v3.Notification.Operation;
+import org.eclipse.uprotocol.v1.UUri;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+public class Notifier {
```

Comment:
Please document your classes so people reading your code will know what they are for

Reply:
Related to my other comment above. I agree commenting classes and functions is a good practice but was not enforced due to lack of a coding standard. Would you like us to comment every class and function?

Reply:
Given that they are hidden and internal to the implimentation of the service, it is less of an issue. BCP though is to document your code :-). Now a days CoPilot will populate the documentation automatically for you.

Reply:
Most of the class document are done. Please review.